### PR TITLE
Add an ability to place heroes in the Editor

### PR DESCRIPTION
**Pull request by ihhub**
This pull request add an ability to place heroes on the Adventure Map within the Editor (relates to #6845). Here are important changes being made:
- move hero rendering code from **heroes_move.cpp** file to **maps_tile_render.h|cpp** files as it is rendering related code, not movement
- GameArea class should know what it is rendering for. Hence, an introduction of `bool isEditor()` method

In the subsequent PR I am going to make heroes rendering as during gameplay: with moving flags and shadows as well as add a logic to ban to place more than 8 heroes. This pull request is fully focused on placing hero "objects" on the Adventure Map.

https://github.com/ihhub/fheroes2/assets/19829520/b6775e5f-31c3-43b4-b3b8-0627afa502ec

### CODE REVIEW

**Comment by Branikolog** [comment]
Hi, @ihhub !

I cannot place a blue knight hero on a map somehow.

Also, I've noticed, that all yellow heroes (except sorceress) have glowing flags:

https://github.com/ihhub/fheroes2/assets/55348946/bb3735a6-b213-4787-b798-0bc6194fafdd

**Comment by ihhub** [comment]
Hi @Branikolog , please ignore flag rendering. This is a problem with the original images.

**Comment by ihhub** [comment]
Hi @Branikolog , I addressed your concern with the first hero. Please test the latest build once it's completed.

File: src/fheroes2/editor/editor_interface_panel.h [hunk 1]
```diff
@@ -204,5 +214,7 @@ namespace Interface
         uint8_t _selectedBrushSize{ BrushSize::MEDIUM };
 
         int32_t _monsterId{ Monster::UNKNOWN };
+
+        int32_t _heroId{ -1 };
```
Comment (Districh-ru) [inline src/fheroes2/editor/editor_interface_panel.h:218]: What do you think of naming it `_heroType`, in example?
When seeing at `heroId` the first is about a specific hero (Lord Kilburn, etc.).

File: src/fheroes2/maps/maps_tiles_helper.h [hunk 2]
```diff
@@ -163,5 +163,8 @@ namespace Maps
     void setTerrainOnTiles( const int32_t startTileId, const int32_t endTileId, const int groundId );
     bool updateRoadOnTile( Tiles & tile, const bool setRoad );
     bool updateStreamOnTile( Tiles & tile, const bool setStream );
+
     void setRandomMonsterOnTile( Tiles & tile, const Monster & mons );
+
+    void setEditorHeroOnTile( Tiles & tile, const int32_t heroId );
```
Comment (Districh-ru) [inline src/fheroes2/maps/maps_tiles_helper.h:169]: The same comment about `heroId` - it is just a type (color+race), but not the ID of a specific hero. :)

**Comment by Branikolog** [comment]
Hi, @ihhub !
Now I can place blue knight hero too!

Between, do we need to restrict the number of placed heroes, or is this out of the current PR?
image

**Review by Districh-ru — APPROVED** [review]
Hi, @ihhub, well done!
I just left two optional suggestions regarding variables naming. Could you please take a look in your free time.
As I understand setting hero's metadata (color, default army and skills) will done in a new PR.

**Comment by ihhub** [comment]
Hi @Branikolog as stated in the description the hero limit is for another pull request.

**Review by Branikolog — APPROVED** [review]

**Comment by ihhub** [comment]
Hi @Districh-ru , I addressed your concerns regarding variable and function names.

**Review by Districh-ru — APPROVED** [review]

**Pull request merged**